pkg/api-server/inspect_endpoints.go

```diff
@@ -313,7 +314,7 @@ func inspectPolicies(
 func routeDestinationToAPIDestination(des route.Destination) api_server_types.Destination {
 	policies := api_server_types.PolicyMap{}
 	for kind, p := range des.Policies {
-		policies[kind] = rest_unversioned.From.Resource(p)
+		policies[kind] = rest.From.Resource(p).GetMeta()
```

Comment:
It seems like an unnecessary conversion of the whole resource and then taking `meta` from the converted result. I'd rather implement a method that converts only `meta`:
```go
func (f *from) ResourceMeta(r core_model.ResourceMeta) v1alpha1.ResourceMeta {
}
```
later we can use `rest.From.ResourceMeta()` inside `rest.From.Resource()`

Reply:
Yes and no. ResourceMeta is not working great here because we need to type which is in the resource descriptor (this is why I introduced From.Meta that should be used here